plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,273 @@
+    T_i={"can_be_negative": False, "equivalencies": u.temperature_energy()},
+    theta={"can_be_negative": True},
+)
+def two_fluid_dispersion_solution(
+    *,
+    B: u.T,
+    k: u.m ** -1,
+    n: u.m ** -3,
+    T_e: u.K,
+    T_i: u.K,
+    gamma_e=1,
+    gamma_i=3,
+    ion="p+",
+    m_e: u.kg = m_e,
+    m_i: u.kg = m_p,
+    theta: u.deg = 45 * u.deg,
+    z=1,
+):
+    r"""
+    Using the solution provided by Bellan 2012, calculate the analytical
+    solution to the two fluid, low-frequency (:math:`\omega/kc \ll 1`) dispersion
+    relation presented by Stringer 1963.  This dispersion relation also
+    assummes a uniform magnetic field :math:`\mathbf{B_o}`, no D.C. electric
+    field :math:`\mathbf{E_o}=0`, and quasi-neutrality.  For more information
+    see the **Notes** section below.
+
+    **Aliases:** `tfds_`
+
+    Parameters
+    ----------
+    B : ~astropy.units.Quantity
+        Magnetic field
+    gamma_e : float or int
+        The adiabatic index for electrons, which defaults to 1.  This
+        value assumes that the electrons are able to equalize their
+        temperature rapidly enough that the electrons are effectively
+        isothermal.
+    gamma_i : float or int
+        The adiabatic index for ions, which defaults to 3. This value
+        assumes that ion motion has only one degree of freedom, namely
+        along magnetic field lines.
+    ion : string, optional
+        Representation of the ion species (e.g., `'p'` for protons,
+        `'D+'` for deuterium, or 'He-4 +1' for singly ionized
+        helium-4), which defaults to protons.  If no charge state
+        information is provided, then the ions are assumed to be
+        singly charged.
+    k : ~astropy.units.Quantity
+        Wave number
+    m_e : ~astropy.units.Quantity
+        Mass of negative ion which defaults to electron
+    m_i : ~astropy.units.Quantity
+        Mass of positive ion which defaults to proton
+    n : ~astropy.units.Quantity
+        Number density.
+    T_e : ~astropy.units.Quantity
+        The electron temperature
+    T_i : ~astropy.units.Quantity
+        The ion temperature
+    theta : ~astropy.units.Quantity
+        Angle of propagation which defaults to 45 degrees.
+    z : float or integer
+        Average ionization number which defaults to 1 for protons.
+
+    Returns
+    -------
+    omega : `~astropy.units.Quantity`
+        A dictionary of Wave frequencies corresponding to three modes:
+        (1) Ion-acoustic mode, (2) Alfven mode, and (3) Fast mode, in 1/s units.
+
+    Raises
+    ------
+    TypeError
+        The magnetic field and density arguments are not instances of
+        `~astropy.units.Quantity` and cannot be converted into those.
+
+    ~astropy.units.UnitConversionError
+        If the magnetic field or density is not in appropriate units.
+
+    ValueError
+        If the magnetic field, density or wavenumber is negative, or the ion
+        mass or charge state cannot be found.
```

Comment:
```suggestion
    TypeError
        If applicable arguments are not instances of `~astropy.units.Quantity` or
        cannot be converted into one.

    TypeError
        If ``ion`` is not of type or convertible to `~plasmapy.particles.Particle`.

    TypeError
        If ``gamma_e``, ``gamma_i``, or``z_mean`` are not of type `int` or `float`.

    ~astropy.units.UnitConversionError
        If applicable arguments do not have units convertible to the expected
        units.

    ValueError
        If the ``B``, ``k``, ``n_i``, ``T_e``, or ``T_i`` is negative.

    ValueError
        If ``ion`` is not of category ion or element.

    ValueError
        If ``B``, ``n_i``, ``T_e``, or ``T_I`` are not single valued
        `astropy.units.Quantity` (i.e. an array).

    ValueError
        If ``k`` or ``theta`` are not single valued or a 1-D array.
```

Update to include all raised `Exception`s introduced by the argument conditioning (see below review comments).